hal/src/rtl872x/usart_hal.cpp

```diff
@@ -831,12 +835,15 @@ class Usart {
                 curTxCount_ = consumable;
                 GDMA_Init(txDmaInitStruct_.GDMA_Index, txDmaInitStruct_.GDMA_ChNum, &txDmaInitStruct_);
                 GDMA_Cmd(txDmaInitStruct_.GDMA_Index, txDmaInitStruct_.GDMA_ChNum, ENABLE);
+                busy_ = true;
+                UART_INTConfig(uartInstance, RUART_IER_ETBEI, ENABLE);
             } else {
                 // LOG UART doesn't support DMA transmission
                 consumable = std::min(MAX_UART_FIFO_SIZE, consumable);
                 auto ptr = txBuffer_.consume(consumable);
                 for (size_t i = 0; i < consumable; i++, ptr++) {
                     UART_CharPut(uartInstance, *ptr);
+                    UART_INTConfig(uartInstance, RUART_IER_ETBEI, ENABLE);
                 }
```

Comment:
Worst case scenario this might trigger for each single byte written here. The below case of enabling this interrupt after this is written is probably correct.

Reply:
Yes, the interrupt might significantly slow down the performance for single byte per transmission.

Reply:
Hmm, actually I don't remember how that got there... I'll remove it from line 846.